client/ayon_core/plugins/publish/integrate_hero_version.py

```diff
@@ -415,11 +423,11 @@ def integrate_instance(
             # Copy(hardlink) paths of source and destination files
             # TODO should we *only* create hardlinks?
             # TODO should we keep files for deletion until this is successful?
-            for src_path, dst_path in src_to_dst_file_paths:
-                self.copy_file(src_path, dst_path)
-
-            for src_path, dst_path in other_file_paths_mapping:
-                self.copy_file(src_path, dst_path)
+            with concurrent.futures.ThreadPoolExecutor() as executor:
+                file_futures = []
+                for src_path, dst_path in src_to_dst_file_paths + other_file_paths_mapping:
+                    file_futures.append(executor.submit(self.copy_file, src_path, dst_path))
+                concurrent.futures.wait(file_futures)
```

Comment:
Since the other code needed it to - I assume this would also need error handling? I'm wondering if it warrants making a small wrapper around `ThreadPoolExecutor` that allows to easily call the list of futures - yet error as soon as one errors along the way.

Preferably we'd not want the rest of the pool to continue execution either from that point onwards.